src/crimson/os/seastore/onode_manager/simple-fltree/onode_delta.h

```diff
@@ -0,0 +1,70 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <cstdint>
+
+#include "common/hobject.h"
+#include "include/buffer_fwd.h"
+
+#include "crimson/os/seastore/onode.h"
+#include "crimson/os/seastore/seastore_types.h"
+
+using crimson::os::seastore::OnodeRef;
+
+struct delta_t {
+  enum class op_t : uint8_t {
+    nop,
+    insert_onode,
+    update_onode,
+    insert_child,
+    update_key,
+    shift_left,
+    trim_right,
+    insert_front,
+    insert_back,
+    remove_from,
+    // finer grained op?
+    //  - changing the embedded extent map of given oid
+    //  - mutating the embedded xattrs of given oid
+  } op = op_t::nop;
+
+  unsigned n = 0;
```

Comment:
It looks that `n` has different meanings depending on `op`. How about emphasizing that with:

```cpp
union {
  unsigned n;
  unsigned slot;
} op_param;
```
?

Reply:
this would fail the build with GCC 10.1, i am afraid. see #35197.